sgm/inference/api.py

```diff
@@ -0,0 +1,391 @@
+            samples,
+            skip_encode=True,
+            return_latents=return_latents,
+            filter=None,
+        )
+
+
+def get_guider_config(params: SamplingParams):
+    if params.guider == Guider.IDENTITY:
+        guider_config = {
+            "target": "sgm.modules.diffusionmodules.guiders.IdentityGuider"
+        }
+    elif params.guider == Guider.VANILLA:
+        scale = params.scale
+
+        thresholder = params.thresholder
+
+        if thresholder == Thresholder.NONE:
+            dyn_thresh_config = {
+                "target": "sgm.modules.diffusionmodules.sampling_utils.NoDynamicThresholding"
+            }
+        else:
+            raise NotImplementedError
+
+        guider_config = {
+            "target": "sgm.modules.diffusionmodules.guiders.VanillaCFG",
+            "params": {"scale": scale, "dyn_thresh_config": dyn_thresh_config},
+        }
+    else:
+        raise NotImplementedError
+    return guider_config
+
+
+def get_discretization_config(params: SamplingParams):
+    if params.discretization == Discretization.LEGACY_DDPM:
+        discretization_config = {
+            "target": "sgm.modules.diffusionmodules.discretizer.LegacyDDPMDiscretization",
+        }
+    elif params.discretization == Discretization.EDM:
+        discretization_config = {
+            "target": "sgm.modules.diffusionmodules.discretizer.EDMDiscretization",
+            "params": {
+                "sigma_min": params.sigma_min,
+                "sigma_max": params.sigma_max,
+                "rho": params.rho,
+            },
+        }
+    else:
+        raise ValueError(f"unknown discertization {params.discretization}")
+    return discretization_config
+
+
+def get_sampler_config(params: SamplingParams):
+    discretization_config = get_discretization_config(params)
+    guider_config = get_guider_config(params)
+    sampler = None
+    if params.sampler == Sampler.EULER_EDM or params.sampler == Sampler.HEUN_EDM:
+        if params.sampler == Sampler.EULER_EDM:
+            sampler = EulerEDMSampler(
+                num_steps=params.steps,
+                discretization_config=discretization_config,
+                guider_config=guider_config,
+                s_churn=params.s_churn,
+                s_tmin=params.s_tmin,
+                s_tmax=params.s_tmax,
+                s_noise=params.s_noise,
+                verbose=True,
+            )
+        elif params.sampler == Sampler.HEUN_EDM:
+            sampler = HeunEDMSampler(
+                num_steps=params.steps,
+                discretization_config=discretization_config,
+                guider_config=guider_config,
+                s_churn=params.s_churn,
+                s_tmin=params.s_tmin,
+                s_tmax=params.s_tmax,
+                s_noise=params.s_noise,
+                verbose=True,
+            )
+    elif (
+        params.sampler == Sampler.EULER_ANCESTRAL
+        or params.sampler == Sampler.DPMPP2S_ANCESTRAL
+    ):

```

Comment:
```suggestion
    elif params.sampler in (Sampler.EULER_ANCESTRAL, Sampler.DPMPP2S_ANCESTRAL):
```
But it's also unnecessary like the above wrapping `if`.